Assets/Scripts/Screeps3D/Rooms/RoomUnpacker.cs

```diff
@@ -120,6 +123,8 @@ private void ControlVisibility(bool showObjects)
         private void UnpackFlags(JSONObject roomData)
         {
             var flagsData = roomData["flags"];
+            flagList.Clear();
+            activeFlagList.Clear();
```

Comment:
flagList is to store the name of all the flag names after the flagstring is split up.  So this will contain all the flagnames unpacked from the JSON.
activeFlagList was to track as I was iterating the flags currently in a room that were valid and part of the JSON payload.

Then if the activeFlagList does not contain a flagname, then we know this is a stale flag and can mark it for removal.